addons/stock/models/stock_orderpoint.py

```diff
@@ -424,7 +424,8 @@ def _get_orderpoint_action(self):
 
         orderpoints = self.env['stock.warehouse.orderpoint'].with_user(SUPERUSER_ID).create(orderpoint_values_list)
         for orderpoint in orderpoints:
-            orderpoint.route_id = orderpoint.product_id.route_ids[:1] or orderpoint._set_default_route_id()
+            orderpoint.route_id = orderpoint.product_id.route_ids.filtered(lambda route: route.supplied_wh_id == orderpoint.location_id.warehouse_id)\
```

Comment:
You are ignoring the routes without any `supplied_wh_id` (for instance, the route Buy) and there will be a bug if `filtered` returns more than one route
I think that something like this could do the job:
```suggestion
            orderpoint_wh = orderpoint.location_id.warehouse_id
            orderpoint.route_id = next((r for r in orderpoint.product_id.route_ids if not r.supplied_wh_id or r.supplied_wh_id == orderpoint_wh), None) \
```